packages/location-field/src/index.tsx

```diff
@@ -645,57 +645,76 @@ const LocationField = ({
     const transitFeaturesPresent =
       stopFeatures.length > 0 || stationFeatures.length > 0;
 
-    const OtherFeaturesHeader = () => (
-      <S.MenuGroupHeader as={headingType} bgColor="#333" key="other-header">
-        <FormattedMessage
-          description="Text for header above the 'other' category of geocoder results"
-          id="otpUi.LocationField.other"
-        />
-      </S.MenuGroupHeader>
-    );
-
-    const otherFeaturesElements = otherFeatures.map(feature =>
-      renderFeature(itemIndex++, feature)
-    );
-
-    // Iterate through the geocoder results
-    menuItems = menuItems.concat(
-      renderOtherFirst &&
-        transitFeaturesPresent &&
-        otherFeatures.length > 0 && <OtherFeaturesHeader />,
-      renderOtherFirst && otherFeaturesElements,
-      stationFeatures.length > 0 && (
-        <S.MenuGroupHeader
-          as={headingType}
-          bgColor={layerColorMap.stations}
-          key="gtfs-stations-header"
-        >
+    const FeaturesElements = ({
+      bgColor,
+      key,
+      titleId,
+      featuresArray
+    }: {
+      bgColor: string;
+      key: string;
+      titleId: string;
+      featuresArray: JSX.Element[];
+    }) => {
+      const Header = () => (
+        <S.MenuGroupHeader as={headingType} bgColor={bgColor} key={key}>
           <FormattedMessage
-            description="Text for header above Stations"
-            id="otpUi.LocationField.stations"
+            description="Text for header above the 'other' category of geocoder results"
+            id={`otpUi.LocationField.${titleId}`}
```

Comment:
I assume this breaks the i18n string checks? Also we've spent a lot of effort and time elsewhere to not do this, are we ok with doing it here?

Reply:
I think there's a way to skip any ids we're dynamically generating in the i18n check, but it doesn't feel worth it here so I passed the entire i18n id instead!